Accounting tests

jjcnn opened the pull request:
Draft PR - not yet ready for merge.

@vaivaswatha has requested that this PR links to https://github.com/Zilliqa/scilla-rtl/issues/46.

jjcnn commented:
@KaustubhShamshery : Any news on the C++ side of the accounting issue?

vaivaswatha commented:
@jjcnn I just merged `master` to this branch. Can you confirm that it's all sane still?

jjcnn commented:
> @jjcnn I just merged `master` to this branch. Can you confirm that it's all sane still?

Everything still looks good. All tests pass, and the code from `master` doesn't seem to touch anything accounting-related.

vaivaswatha commented:
>Additional accounting tests

@jjcnn `AcceptAndCheckSenderBalance` in the support contract doesn't take an `expected_balance` parameter, but this commit assumes so.

jjcnn commented:
> > Additional accounting tests
> 
> @jjcnn `AcceptAndCheckSenderBalance` in the support contract doesn't take an `expected_balance` parameter, but this commit assumes so.

@vaivaswatha : Fixed.

I've also fixed the merge conflicts, so the branch should be in order again.
